tests/tests_e2e/ustream_aux_e2e/ustream_aux_e2e.c

```diff
@@ -210,6 +216,252 @@ TEST_FUNCTION(az_ustream_multi_read_concurrent_succeed)
+    size_t remaining_from_second = strlen((const char *)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_2) - read_length;
+    //read starting from the second expected content + what was read in previous ustream
+    result = az_ustream_read(&ustream_instance_split, buf_result, remaining_from_second, &returned_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, 0, strncmp((const char*)buf_result, (const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_2 + read_length,
+                                                       returned_size));
+
+    memset(&buf_result, 0, USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH);
+    //read through third expected content
+    read_length = strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_3);
+    result = az_ustream_read(&ustream_instance_split, buf_result, read_length, &returned_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, 0, strncmp((const char*)buf_result, (const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_3,
+                                                       strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_3)));
+
+    result = az_ustream_get_remaining_size(&ustream_instance_split, &remaining_size);
+    ASSERT_ARE_EQUAL(int, 0, remaining_size);
+
+    ///cleanup
+    free(test_buf);
+    az_ustream_dispose(&ustream_instance);
+    az_ustream_dispose(&ustream_instance_split);
+}
+
+TEST_FUNCTION(az_ustream_split_concat_and_split_and_dispose_succeed)
+{
+    ///arrange
+    AZ_USTREAM ustream_instance;
+   
+    create_test_default_multibuffer(&ustream_instance);
+
+    uint8_t buf_result[USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH] = { 0 };
+    size_t split_location = USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH / 2;
+    size_t returned_size;
+
+    AZ_USTREAM ustream_instance_split;
+
+    AZ_ULIB_RESULT result = az_ustream_split(&ustream_instance, &ustream_instance_split, split_location);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+
+    ///act
+    //read the ustream in two parts to ensure properly split
+    size_t read_length;
+    read_length = strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_1);
+    result = az_ustream_read(&ustream_instance, buf_result, read_length, &returned_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, 0, strncmp((const char*)buf_result, (const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_1,
+                                                       strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_1)));
+    memset(&buf_result, 0, USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH);
+    read_length = split_location - read_length;
+    az_ustream_dispose(&ustream_instance_split);
+
+    //assert
+    result = az_ustream_read(&ustream_instance, buf_result, read_length, &returned_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    uint8_t* test_buf = (uint8_t*)malloc(read_length);
```

Comment:
>uint8_t* test_buf = (uint8_t*)malloc(read_length); [](start = 4, length = 50)

Consider move it to the stack to avoid leak in case of the assert fail.

Reply:
Like buf_result

---
In reply to: [334064850](https://github.com/Azure/azure-ulib-c/pull/36#discussion_r334064850) [](ancestors = 334064850)

Reply:
i had that originally but the windows build fails because it doesn't like buffers on the stack with unknown sizes

---
In reply to: [334065133](https://github.com/Azure/azure-ulib-c/pull/36#discussion_r334065133) [](ancestors = 334065133,334064850)